ocaml/xapi/certificates_sync.ml

```diff
@@ -0,0 +1,87 @@
+module D = Debug.Make (struct let name = "certificates_sync" end)
+
+open D
+module Unixext = Xapi_stdext_unix.Unixext
+module Date = Xapi_stdext_date.Date
+module G = Gencertlib
+module GP = Gencertlib.Pem
+open Api_errors
+open Rresult
+
+let ( let* ) = ( >>= )
+
+let uninstall ~__context cert =
+  info "Removing certificate %s" (Ref.string_of cert) ;
+  Db.Certificate.destroy ~__context ~self:cert
+
+(* the host cert is under a fixed name - so the new cert in the file
+   system already replaced the old one. We must not remove it from the
+   file system *)
+
+(** add a certificate to the database. The certificate is already in the
+  file system. This creates a new entry in the database. *)
+let install ~__context ~host cert =
+  try
+    let ref =
+      Certificates.Db_util.add_cert ~__context ~type':(`host host) cert
+    in
+    info "Adding host certificicate %s to database" (Ref.string_of ref) ;
+    R.ok ()
+  with e ->
+    error "certificates_sync.install exception: %s" (Printexc.to_string e) ;
+    Error (`Msg ("installation of host certificate failed", []))
+
+(** determine if the database is up to date by comparing the fingerprint
+  of xapi-ssl.pem with the entry in the database *)
+let is_unchanged ~__context cert_ref cert =
+  let ref_hash = Db.Certificate.get_fingerprint ~__context ~self:cert_ref in
+  let cert_hash =
+    X509.Certificate.fingerprint Mirage_crypto.Hash.(`SHA256) cert
+    |> Certificates.pp_hash
+  in
+  Ok (cert_hash = ref_hash)
```

Comment:
The function can't return an error, it's worth dropping the result here.
```suggestion
  cert_hash = ref_hash
```

Reply:
This needs a change at the caller, too. I've added it.